server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES85BloomFilterPostingsFormat.java

```diff
@@ -322,13 +320,22 @@ static final class FieldsReader extends FieldsProducer {
                 );
                 CodecUtil.retrieveChecksum(indexIn);
                 success = true;
+                assert assertBloomFilterSizes(state.segmentInfo);
```

Comment:
Seems a bit odd to count an assertion failure as `success` here - maybe switch these lines?

```suggestion
                assert assertBloomFilterSizes(state.segmentInfo);
                success = true;
```